File: mt9x9.java
/*
9x9 multiplication table in Java
CC0, Wei-Lun Chao <bluebat@member.fsf.org>, 2018.
*/
// javac mt9x9.java && java mt9x9

public class mt9x9 {
    public static void main(String args[]) {
        for(int i = 1; i <= 9; i += 3) {
            for(int j = 1; j <= 9; j++) {
                for(int k: new int[]{i, i+1, i+2})
                    System.out.printf("%dx%d=%2d\t", k, j, k*j);
                System.out.printf("\n");
            }
            System.out.println();
        }
    }
}

File: mt9x9.hs
{-
9x9 multiplication table in Haskell
CC0, Wei-Lun Chao <bluebat@member.fsf.org>, 2019.
-}
-- runghc mt9x9.hs || ( ghc mt9x9.hs ; ./mt9x9 )

import Control.Monad
main = do
    forM_ [1,4..9] $ \i -> do
        forM_ [1..9] $ \j -> do
            forM_ [i, i+1, i+2] $ \k -> do
                putStr (show k ++ "x" ++ show j ++ "=" ++
                    (if (k*j) < 10 then " " else "") ++
                    show (k*j) ++ "\t")
            putStrLn ""
        putChar '\n'
